### DISCUSSION TITLE
Draw Fixed-Sized length line from x1,y1,x2,y2

### DISCUSSION THREAD

**Discussion by ghost** [fabricjs/fabric.js#7658] [Q&A]
I found a way to draw `x%` of a line from another line.

```
// x1,y1,x2,y2 are points of original line.
var anotherLine = new fabric.Line([x1, y1, x1 + ((x2 - x1) / 5), y1 + ((y2 - y1) / 5)] // draws 20% of line since we divide by four.
```

But I want to draw a fixed-sized line (not percentage) from the original line's points. I'm unable to find the calculation method for this. Any help/guidance would be highly appreciated. Thanks!

**Comment by ShaMan123** [comment]
Line is a bit strange IMO. Did you try Path/Polyline?

**Reply by ghost** [reply]
Hi, Thanks for the reply! I tried this answer: https://stackoverflow.com/a/41729978

It worked for me. Need to check the edge cases.
